### PR TITLE
Added service_configs map to server config

### PR DESCRIPTION
**Pull request by mangchiandjjoe**

### CODE REVIEW

File: contrib/endpoints/src/api_manager/proto/server_config.proto [hunk 1]
```diff
@@ -44,6 +44,12 @@ message ServerConfig {
   // The service name.
   string service_name = 8;
 
+  // Maps service configuration files to their corresponding traffic percentage.
+  // Key is the service configuration file, Value is the traffic percentage
+  // which must be greater than 0.0 and the sum must equal to 100.0.
+  map<string, double> service_configs = 12;
```
Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/proto/server_config.proto]: I prefer to define a message and use repeated field.  call it 

init_service_configs

Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/proto/server_config.proto]: also describe the relationship between this and config_id

File: contrib/endpoints/src/api_manager/proto/server_config.proto [hunk 2]
```diff
@@ -196,6 +194,15 @@ message ServiceManagementConfig {
   int32 refresh_interval_ms = 2;
 }
 
+// Maps service configuration files to their corresponding traffic percentage.
+message ServiceConfigTrafficPercentage {
+  // The service configuration file name
+  string service_config = 1;
```
Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/proto/server_config.proto]: call it service_config_file_full_path

File: contrib/endpoints/src/api_manager/proto/server_config.proto [hunk 3]
```diff
@@ -44,15 +44,13 @@ message ServerConfig {
...
-  map<string, double> service_configs = 12;
-
-  // TODO(jaebong): This will be deprecated once service_configs is applied
-  // The service config id to use at the start of ESP.
+  // The service config id to use at the start of ESP. It has higher precedence
+  // than init_service_configs
   string config_id = 9;
 
+  // Init Service Configs.
+  repeated ServiceConfigTrafficPercentage init_service_configs = 12;
```
Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/proto/server_config.proto]: add
// only use when service_id is not specified and not in the metadata either.

File: contrib/endpoints/src/api_manager/proto/server_config.proto [hunk 4]
```diff
@@ -196,6 +194,15 @@ message ServiceManagementConfig {
   int32 refresh_interval_ms = 2;
 }
 
+// Maps service configuration files to their corresponding traffic percentage.
+message ServiceConfigTrafficPercentage {
+  // The service configuration file name
+  string service_config = 1;
+  // The traffic percentage which must be greater than 0.0 and the sum must
+  // equal to 100.0.
```
Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/proto/server_config.proto]: no, we don't need to add such restriction.  let is call it 

traffic_weight.   

the traffic is spitted according to the weights. sum of them not need to be 100%

Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/proto/server_config.proto]: internally was implemented as weight.

File: contrib/endpoints/src/api_manager/proto/server_config.proto (hunk 3 above)
Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/proto/server_config.proto]: Actually, it should be:
init_service_configs first, 

config_id is not supported,  it require init_loading,   we don't support it,  it should be removed.

File: contrib/endpoints/src/api_manager/proto/server_config.proto (hunk 4 above)
Comment (mangchiandjjoe) [inline contrib/endpoints/src/api_manager/proto/server_config.proto]: Can we ignore this comment?

    // Maps service configuration IDs to their corresponding traffic percentage.
    // Key is the service configuration ID, Value is the traffic percentage
    // which must be greater than 0.0 and the sum must equal to 100.0.

https://github.com/googleapis/googleapis/blob/master/google/api/servicemanagement/v1/resources.proto#L216

File: contrib/endpoints/src/api_manager/proto/server_config.proto (hunk 2 above)
Comment (mangchiandjjoe) [inline contrib/endpoints/src/api_manager/proto/server_config.proto]: Renamed

File: contrib/endpoints/src/api_manager/proto/server_config.proto (hunk 3 above)
Comment (mangchiandjjoe) [inline contrib/endpoints/src/api_manager/proto/server_config.proto]: Added

Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/proto/server_config.proto]: Now, init_config is from service_content passed to construct of esp,  or this one.  That is it.

Ideally, we should change passing service_content to the ESP constructor to server_config too.

File: contrib/endpoints/src/api_manager/proto/server_config.proto [hunk 5]
```diff
@@ -44,11 +44,8 @@ message ServerConfig {
   // The service name.
   string service_name = 8;
 
-  // The service config id to use at the start of ESP. It has higher precedence
-  // than init_service_configs
-  string config_id = 9;
-
   // Init Service Configs.
+  // only use when service_id is not specified and not in the metadata either.
```
Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/proto/server_config.proto]: remove this comment too

File: contrib/endpoints/src/api_manager/proto/server_config.proto [hunk 6]
```diff
@@ -44,8 +44,8 @@ message ServerConfig {
   // The service name.
   string service_name = 8;
```
Comment (qiwzhang) [inline contrib/endpoints/src/api_manager/proto/server_config.proto]: I think we can remove "service_name" too.

Comment (mangchiandjjoe) [inline contrib/endpoints/src/api_manager/proto/server_config.proto]: Removed

**Pull request merged**